ci/run.sh

```diff
@@ -2,6 +2,40 @@
 
 set -ex
 
+# install cmake. cmake required for zlib-ng build, and old cmake may not work,
+# so we install the latest one. this step only run for docker tests.
+if [ -f /.dockerenv ]; then
+    CMAKE_URL=https://github.com/Kitware/CMake/releases/download/v3.25.0/cmake-3.25.0-linux-x86_64.tar.gz
```

Comment:
This is only done in a docker environment so could this be part of the container build steps? Most of the curl/wget/apt-get stuff shouldn't be necessary since the container can just install one and run with that.

Reply:
There are 6 different docker images here, and they are based on centos:7 and ubuntu:16.04 respectively, I think if we do this step in the Dockerfile, there will be 6 copies of almost the same script code.

Reply:
You can add a `*.sh` script and then run the script in the container as well which should deduplicate this.

Reply:
Ok let me change it